specification/src/main/asciidoc/jakarta-concurrency.adoc

```diff
@@ -2302,3 +2302,159 @@ ends.
+         PreparedStatement stmt = con.prepareStatement(CHECK_AVAILABILITY);
+         for (Item item : combined) {
+            ... Remove if the similar item is unavailable
+         }
+      } catch (NamingException | SQLException x) {
+         throw new CompletionException(x);
+      }
+      return Async.Result.complete(combined);
+   }
+}
+----
+====== Asynchronous Method Invocation
+
+The CDI managed bean with the asynchronous method is injected into a
+a `Servlet`, which uses it to asynchronously determine the product
+recommendations.
+
+[source,java]
+----
+public class CheckoutServlet extends HttpServlet {
+   @Inject
+   ProductRecommendations recommendations;
+
+   @Resource(name=”java:comp/env/jdbc/ds1”, lookup="jdbc/ds1")
+   DataSource ds;
+
+   public void doGet(HttpServletRequest req, HttpServletResponse resp)
+      throws ServletException {
+      ...
+      recommendations.findSimilar(cust.getCart(), cust.getHistory())
+                     .thenAccept(recommended -> {
+                        ... Update page with recommendations
+                     });
+      ...
+   }
+}
+----
+
+==== Jakarta EE Product Provider’s Responsibilities
+
+The Jakarta EE Product Provider’s responsibilities are as defined in
+EE.5.8.3.
+
+The Jakarta EE Product Provider registers a CDI interceptor
+to arrange for the invocation of asynchronous methods on the
+`jakarta.enterprise.concurrent.ManagedExecutorService` or
+`jakarta.enterprise.concurrent.ManagedScheduledExecutorService`
+that is specified by the `jakarta.enterprise.concurrent.Async`
+annotation.
+
+The Jakarta EE Product Provider creates a
+`java.util.concurrent.CompletableFuture` instance to associate with each
+asynchronous method invocation, returning this same instance to the caller
+of the asynchronous method, and providing it to the asynchronous method
+implementation by means of the
+`jakarta.enterprise.concurrent.Async.Result` API. The Jakarta EE Product
+Provider completes this instance upon completion of the completion stage
+that is returned by the asynchronous method implementation, which is a
+no-op when the asynchronous method implementation chooses to return the
+same instance. If the asynchronous method return type is `void` or if the
+asynchronous method implementation raises an error or exception, the
+Jakarta EE Product Provider completes this instance upon return from the
+asynchronous method implementation. The Jakarta EE Product Provider
+raises `java.util.concurrent.RejectedExecutionException` to the caller of
+the asynchronous method if it cannot accept a method for asynchronous
+execution, for example if supplied with an invalid JNDI name.
+If the Jakarta EE Product Provider cannot start the asynchronous method
+for any reason after this point, it completes the
+`java.util.concurrent.CompletableFuture` instance with a
+`java.util.concurrent.CancellationException`.
+
+==== Transaction Management
+
+Asynchronous method can be annotated with
```

Comment:
This bit is unclear to me. If the Method is marked `jakarta.transaction.Transactional.TxType.REQUIRES_NEW` and `Asynch` is the asynch invocation in a new transaction, my reading is that is implied but not explicitly stated? Also what happen if it is marked with `REQUIRED`

Reply:
Good point -  `@Transactional` with `REQUIRES_NEW` will require the method to run in a new transaction, and we should be explicit about that. I'll add that in for clarity after we get the rest of this figured out.  Regarding what happens with `REQUIRED`, I documented that on the `@Async` API, `All other transaction attributes must result in {@link java.lang.UnsupportedOperationException UnsupportedOperationException} upon invocation of the asynchronous method.` but I forgot to also include that here.  Thanks for spotting this - I'll plan to get that added in as well.